components/match2/wikis/tft/match_summary.lua

```diff
@@ -0,0 +1,315 @@
+---
+-- @Liquipedia
+-- wiki=tft
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Abbreviation = require('Module:Abbreviation')
+local Class = require('Module:Class')
+local DateExt = require('Module:Date/Ext')
+local Icon = require('Module:Icon')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+local Table = require('Module:Table')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper')
+local MatchSummary = Lua.import('Module:MatchSummary/Base')
+
+local MAP_VETO_START = '<b>Start Map Veto</b>'
+local ARROW_LEFT = '[[File:Arrow sans left.svg|15x15px|link=|Left team starts]]'
+local ARROW_RIGHT = '[[File:Arrow sans right.svg|15x15px|link=|Right team starts]]'
+local NONE = '-'
+local TBD = Abbreviation.make('TBD', 'To Be Determined') --[[@as string]]
+
+---@enum TftMatchIcons
+local Icons = {
+	CHECK = Icon.makeIcon{iconName = 'winner', color = 'forest-green-text', size = '110%'},
+	EMPTY = '[[File:NoCheck.png|link=]]',
+}
+
+local LINK_DATA = {
+	preview = {icon = 'File:Preview Icon32.png', text = 'Preview'},
+}
+
+local VETO_TYPE_TO_TEXT = {
+	ban = 'BAN',
+	pick = 'PICK',
+	decider = 'DECIDER',
+	defaultban = 'DEFAULT BAN',
+
+}
+local CustomMatchSummary = {}
+
+-- Map Veto Class
+---@class TftMapVeto: MatchSummaryRowInterface
+---@operator call: self
+---@field root Html
+---@field table Html
+local MapVeto = Class.new(
+	function(self)
+		self.root = mw.html.create('div'):addClass('brkts-popup-mapveto')
+		self.table = self.root:tag('table')
+			:addClass('wikitable-striped'):addClass('collapsible'):addClass('collapsed')
+		self:createHeader()
+	end
```

Comment:
there is no "standardized" one via commons fwiw
maybe it would be a good idea to push the "default" mapVeto display to commons though
but would consider it oos
if wanted let me know and i can take a look at it later (preferably after this PR here got merged)

copied the R6 one over 1:1 for now